src/Equilibrium/InverseEquilibrium.jl

```diff
@@ -0,0 +1,248 @@
+    newq0 = equil_params.newq0 # New target on-axis safety factor (q0)
+    power_bp = equil_params.power_bp # Exponent for poloidal field weighting
+    power_b = equil_params.power_b # Exponent for total magnetic field weighting
+    power_r = equil_params.power_r # Exponent for major radius weighting
+
+    # 1.2 Set flags
+    power_flag = false #equil_params.power_flag
+    direct_flag = false #equil_params.direct_flag
+    out_eq_1d = false # Enable ASCII output for 1D input data
+    bin_eq_1d = false # Enable binary output for 1D input data
+    out_eq_2d = false # Enable ASCII output for 2D input data
+    bin_eq_2d = false # Enable binary output for 2D input data
+    input_only = false # If true, stop after writing input data
+    interp = false # If true, interpolate data before writing output
+
+    # 1.3 Other unimplemented parameters
+    sp_pfac = 1 # (NEEDS IMPLEMENTATION) Packing factor for "mypack" grid_type
+    jaceq = 1 # (NEEDS IMPLEMENTATION), DIMENSION(:,:), INTENT(IN): Jacobian from CHEASE, for inverse_chease4_run(jaceq)
+
+    # 2. Setup the new output radial grid (`ψ_norm`)
+    mtheta = equil_params.mtheta # Number of poloidal grid points
+    grid_type = equil_params.grid_type # New radial grid type
+    mpsi = equil_params.mpsi # Number of radial grid points
+    psilow = equil_params.psilow # Lower bound of normalized poloidal flux for the new grid
+    psihigh = equil_params.psihigh # Upper bound of normalized poloidal flux for the new grid
+
+    sq_x_nodes = zeros(Float64, mpsi + 1)
+
+    if grid_type == "ldp"
+        for i in 0:mpsi
+            x_param = Float64(i) / mpsi
+            sq_x_nodes[i+1] = psilow + (psihigh - psilow) * sin(x_param * pi / 2.0)^2
+        end
+    else
+        error("Unsupported grid_type: $(equil_params.grid_type)")
+    end
+    # need to add more grid type
+
+    #2pi*F, mu0P, dvdpsi, q
+    sq_fs_nodes = zeros(Float64, mpsi + 1, 4)
+
+    # (Should make this function so both inverse and direct can use it instead of copy-pasting)
+    # Helper function for robust Newton-Raphson search with restarts
+    max_newton_iter = 200
+    function find_separatrix_crossing(start_r, end_r, label)
+        println("Finding $label separatrix crossing...")
+        local r_sol::Float64
+        found = false
+        for ird in 0:5 # 6 restart attempts
+            r_sep = (start_r * (3.0 - 0.5 * ird) + end_r) / (4.0 - 0.5 * ird)
+            @printf "  Restart attempt %d/6 with initial R = %.6f\n" (ird + 1) r_sep
+            for _ in 1:max_newton_iter
+                
+                direct_get_bfield!(bf_temp, r_sep, zo, psi_in_new, sq_in, psio, derivs=1)
+                if abs(bf_temp.psir) < 1e-14; @warn "d(psi)/dr is near zero."; break; end
+                dr = -bf_temp.psi / bf_temp.psir
+                r_sep += dr
+                if abs(dr) <= 1e-12 * abs(r_sep); r_sol = r_sep; found = true; break; end
```

Comment:
I'd avoid these multiline if statements rolled onto one line. It's hard to follow.